File: LeetCode/Python/Easy/349-Intersection-Of-Two-Arrays/solution.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result: list[int]=list()

        intersection: set[int]=set()

        nums2_set: set[int]=set(nums2)

        for number in nums1:
            if number in nums2_set and number not in intersection:
                intersection.add(number)

        result=[number for number in intersection]

        return result

File: LeetCode/Rust/Easy/349-Intersection-of-Two-Arrays/solution.rs
use std::collections::{HashSet};

impl Solution
{
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32>
    {
        let mut result: Vec<i32>=Vec::new();

        let nums2_hashset: HashSet<i32>=nums2.iter().copied().collect::<HashSet<i32>>();

        let mut intersection: HashSet<i32>=HashSet::new();

        for number in nums1.iter()
        {
            if nums2_hashset.contains(number) && !intersection.contains(number)
            {
                intersection.insert(*number);

                result.push(*number);
            }
        }

        result
    }
}